lib/core/sdam/server.js

```diff
@@ -474,7 +493,19 @@ function executeWriteOperation(args, options, callback) {
 }
 
 function connectEventHandler(server) {
-  return function(pool, conn) {
+  return function(err, conn) {
+    if (server.s.state === STATE_CLOSING || server.s.state === STATE_CLOSED) {
```

Comment:
Do we need to clean up anything for the newly connected connection?

Reply:
This is also handled by `DRAINING`. `createConnection` stores the connection in the `availableConnections` local array, so if we're `CLOSING` or `CLOSED` here, the connection is either about to be destroyed or already destroyed - we just need to ignore the event.